extras/wait-for-bricks.sh

```diff
@@ -0,0 +1,156 @@
+                echo "N"
+                ;;
+        1|on|yes|true|enable)
+                echo "Y"
+                ;;
+        *)
+                echo "N/A"
+                ;;
+        esac
+}
+
+function volinfo_option()
+{
+    local vol=$1;
+    local key=$2;
+
+    gluster volume get "$vol" "$key"| awk '{print $NF}' | tail -1
+}
+
+function check_brick_status() {
+        local volname=$1
+        local daemon=$2
+
+        cmd="gluster --xml volume status $volname"
+        if [[ -z $daemon ]]
+         then
+                 $cmd | grep -c '<status>1'
+        else
+                 $cmd | sed -n -e "/${daemon}/,/<status>/ p" | grep -c '<status>1'
+         fi
+}
+
+function volume_online_brick_count()
+{
+        local volname=$1
+        local v1=0
+        local v2=0
+        local v3=0
+        local v4=0
+        local v5=0
+        local tot=0
+
+        #First count total Number of bricks and then subtract daemon status
+        v1=$(check_brick_status "$volname")
+        v2=$(check_brick_status "$volname" "Self-heal")
+        v3=$(check_brick_status "$volname" "Quota")
+        v4=$(check_brick_status "$volname" "Snapshot")
+        v5=$(check_brick_status "$volname" "Tier")
+        tot=$((v1-v2-v3-v4-v5))
+        echo $tot
+}
+
+function volumes_waiting_for_bricks_up()
+{
+        local wait_info
+        local vol_down_info
+        while read -r volname;
+        do
+                if [[ "Started" == $(volinfo_field "$volname" 'statusStr') ]];
+                then
+                        if [[ "Y" == "$(is_enabled "$(volinfo_option "$volname" features.shard)")" ]];
+                        then
+                                brick_count=$(volume_online_brick_count "$volname")
+                                total_brick_count=$(volinfo_field "$volname" 'brickCount')
+                                if [[ $brick_count -ne $total_brick_count ]];
+                                then
+                                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
+                                        if [[ -z "$wait_info" ]];
+                                        then
+                                                wait_info="$vol_down_info"
+                                        else
+                                                wait_info="$wait_info, $vol_down_info"
+                                        fi
+                                fi
+                        fi
+                fi
+        done < <(gluster volume list)
+        if [[ ! -z "$wait_info" ]]; then echo "$wait_info"; fi
+}
+
+function check_glusterd_running()
+{
+        if ! pidof glusterd > /dev/null 2>&1 ; then
+                printLog "ERROR: Glusterd is not running";
+                exit 1;
+        fi
+}
+
+function wait_for_bricks_up()
+{
+        local endtime=$1
+        local now
+
+        while [[ ! -z "$(volumes_waiting_for_bricks_up)" ]]; do
```

Comment:
Same as with if/then, decide if while/do should be on the same line or not, and stick to is for the whole script.